packages/angular/ssr/src/app.ts

```diff
@@ -52,7 +63,44 @@ export class AngularServerApp {
    *
    * @returns A promise that resolves to the HTTP response object resulting from the rendering, or null if no match is found.
    */
-  async render(
+  render(
+    request: Request,
+    requestContext?: unknown,
+    serverContext: ServerRenderContext = ServerRenderContext.SSR,
+  ): Promise<Response | null> {
+    return Promise.race([
+      this.createAbortPromise(request),
+      this.handleRendering(request, requestContext, serverContext),
+    ]);
+  }
+
+  /**
+   * Creates a promise that rejects when the request is aborted.
+   *
+   * @param request - The HTTP request to monitor for abortion.
+   * @returns A promise that never resolves but rejects with an `AbortError` if the request is aborted.
+   */
+  private createAbortPromise(request: Request): Promise<never> {
+    return new Promise<never>((_, reject) => {
+      request.signal.onabort = function () {
```

Comment:
Suggestion: Prefer `addEventListener('abort')` so as not to clobber any existing callback on `onabort`. You can also use `once` for slightly better memory management (probably doesn't matter much in practice here).

```typescript
request.signal.addEventListener('abort', () => {
  // ...
}, {once: true});
```